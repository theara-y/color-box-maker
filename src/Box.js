const Box = ({ testid, removeBox, idx, width, height, color }) => {
    const handleClick = () => {
        removeBox(idx);
    };

    return (
        <>
        <div className="Box" data-testid={testid} style={
            {
                width: width + "px",
                height: height + "px",
                backgroundColor: color
            }
        }></div>
        <button onClick={handleClick}>X</button>
        </>
    );
};

export default Box;