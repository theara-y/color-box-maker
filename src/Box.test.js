import { render } from "@testing-library/react";
import Box from "./Box";

test('render Box', () => {
    render(<Box/>);
});

test('Box snapshot', () => {
    const { asFragment } = render(<Box/>);
    expect(asFragment()).toMatchSnapshot()
});