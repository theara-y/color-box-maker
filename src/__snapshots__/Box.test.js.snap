// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  />
  <button>
    X
  </button>
</DocumentFragment>
`;
